5.0: Update field type reference (#2795)
ibexa/documentation-developer: docs/content_management/field_types/field_type_reference/productspecificationfield.md

---
edition: headless
month_change: true
---

# Product specification field type

This field represents and handles [product attributes](products.md#product-attributes) and [VAT](prices.md#vat).

> Consider it as internal to the [PIM](pim.md). 

Reviewer: ```suggestion
Consider it as internal to the [PIM](pim.md).
```

| Name                   | Internal name                 | Expected input |
|------------------------|-------------------------------|----------------|
| `ProductSpecification` | `ibexa_product_specification` | mixed          |

!!! caution

>     The presence of a specification (`ibexa_product_specification`) field is what distinct product type from content type.

Reviewer: ```suggestion
    The presence of a specification (`ibexa_product_specification`) field distincts product types from content types.
```

    Don't remove this field from a product type (or it becomes a unreachable hidden content type).
    Don't add such field to a content type (or it becomes an uneditable unusable product type).

>     You can't have more than one `ibexa_product_specification`, the laters won't be editable.

Reviewer: ```suggestion
```
I don't think that's possible anyway, that field is marked as singular:
https://github.com/ibexa/product-catalog/blob/main/src/lib/FieldType/ProductSpecification/Type.php#L71-L74

Author: I'm pretty sure I manage to add a second one while testing for the PIM training. Anyway, I didn't manage to do it again. But I found a UI error.

It calls something like /api/ibexa/v2/contenttypegroup/4/contenttype/55/add_field_definition/eng-GB which responses an XML error.

````xml
<?xml version="1.0" encoding="UTF-8"?>
<ErrorMessage media-type="application/vnd.ibexa.api.ErrorMessage+xml">
  <errorCode>409</errorCode>
  <errorMessage>Conflict</errorMessage>
  <errorDescription>Argument '$contentTypeDraft' has a bad state: The content type already contains a Field definition of the singular Field Type 'ibexa_product_specification'</errorDescription>
````

![image](https://github.com/user-attachments/assets/0b6d1137-7bd8-4436-8215-7c1f9fb9ba94)
